pkg/apiserver/authenticator/authn.go

```diff
@@ -96,12 +99,20 @@ func New(config AuthenticatorConfig) (authenticator.Request, error) {
 		authenticators = append(authenticators, serviceAccountAuth)
 	}
 
-	if len(config.KeystoneURL) > 0 {
-		keystoneAuth, err := newAuthenticatorFromKeystoneURL(config.KeystoneURL)
-		if err != nil {
-			return nil, err
+	if len(config.KeystoneConfig) > 0 {
+		if config.KeystoneAuthMode == "token" {
+			keystoneTokenAuth, err := newTokenAuthenticatorFromKeystoneConfig(config.KeystoneConfig)
+			if err != nil {
+				return nil, err
+			}
+			authenticators = append(authenticators, keystoneTokenAuth)
+		} else if config.KeystoneAuthMode == "password" {
+			keystonePasswordAuth, err := newPasswordAuthenticatorFromKeystoneURL(config.KeystoneURL)
```

Comment:
Your right. This is a left over from when I tried to unify the two configs but then was asked to not do that in this patchset. I'll fix it.